tests/test_mongodb_log_based_interruptible.py

```diff
@@ -0,0 +1,330 @@
+
+            # simple_coll_2 has 20 documents
+            client["simple_db"]["simple_coll_2"].insert_many(generate_simple_coll_docs(20))
+
+    def expected_check_streams(self):
+        return {
+            'simple_db-simple_coll_1',
+            'simple_db-simple_coll_2',
+        }
+
+    def expected_pks(self):
+        return {
+            'simple_coll_1': {'_id'},
+            'simple_coll_2': {'_id'},
+        }
+
+    def expected_row_count_1(self):
+        return {
+            'simple_coll_1': 10,
+            'simple_coll_2': 20,
+        }
+
+    def expected_row_count_2(self):
+        return {
+            'simple_coll_1': 3,
+            'simple_coll_2': 13
+        }
+
+    def expected_row_count_3(self):
+        return {
+            'simple_coll_1': 0,
+            'simple_coll_2': 1
+        }
+
+    def expected_sync_streams(self):
+        return {
+            'simple_coll_1',
+            'simple_coll_2'
+        }
+
+    def name(self):
+        return "tap_tester_mongodb_log_based_interruptible"
+
+    def tap_name(self):
+        return "tap-mongodb"
+
+    def get_type(self):
+        return "platform.mongodb"
+
+    def get_credentials(self):
+        return {'password': os.getenv('TAP_MONGODB_PASSWORD')}
+
+    def get_properties(self):
+        return {'host': os.getenv('TAP_MONGODB_HOST'),
+                'port': os.getenv('TAP_MONGODB_PORT'),
+                'user': os.getenv('TAP_MONGODB_USER'),
+                'database': os.getenv('TAP_MONGODB_DBNAME')
+                }
+
+    def test_run(self):
+
+        conn_id = connections.ensure_connection(self)
+
+        #  -------------------------------
+        # -----------  Discovery ----------
+        #  -------------------------------
+
+        # run in discovery mode
+        check_job_name = runner.run_check_mode(self, conn_id)
+
+        # verify check exit codes
+        exit_status = menagerie.get_exit_status(conn_id, check_job_name)
+        menagerie.verify_check_exit_status(self, exit_status, check_job_name)
+
+        # verify the tap discovered the right streams
+        found_catalogs = menagerie.get_catalogs(conn_id)
+
+        # assert we find the correct streams
+        self.assertEqual(self.expected_check_streams(),
+                         {c['tap_stream_id'] for c in found_catalogs})
+
+        for tap_stream_id in self.expected_check_streams():
+            found_stream = [c for c in found_catalogs if c['tap_stream_id'] == tap_stream_id][0]
+
+            # assert that the pks are correct
+            self.assertEqual(self.expected_pks()[found_stream['stream_name']],
+                             set(found_stream.get('metadata', {}).get('table-key-properties')))
+
+            # assert that the row counts are correct
+            self.assertEqual(self.expected_row_count_1()[found_stream['stream_name']],
+                             found_stream.get('metadata', {}).get('row-count'))
```

Comment:
I think these should be covered by the discovery test.

Reply:
yes, so removing it from this test